core/src/main/web/app/mainapp/mainapp.js

```diff
@@ -1160,34 +1160,30 @@
           }
         });
 
-        $scope.promptToSave = (function() {
-          var prompted = false;
-          return function() {
-            if (prompted) { // prevent prompting multiple at the same time
-              return;
-            }
-            prompted = true;
-            bkCoreManager.show2ButtonModal(
-              "Beaker server disconnected. Further edits will not be saved.<br>" +
-              "Save current notebook as a file?",
-              "Disconnected", function() {
-                // "Save", save the notebook as a file on the client side
-                bkSessionManager.dumpDisplayStatus();
-                var timeoutPromise = $timeout(function() {
-                  bkUtils.saveAsClientFile(
-                      bkSessionManager.getSaveData().notebookModelAsString,
-                  "notebook.bkr");
-                }, 1);
-                timeoutPromise.then(function() {
-                  prompted = false;
-                })
-              }, function() {
-                prompted = false;
-              },
-              "Save", "Not now", "btn-primary", ""
-            );
-          };
-        })();
+        $scope.promptToSave = function() {
+          if ($scope.disconnectedDialog) { // prevent prompting multiple at the same time
+            return;
+          }
+          $scope.disconnectedDialog = bkCoreManager.show2ButtonModal(
+            "Beaker server disconnected. Further edits will not be saved.<br>" +
+            "Save current notebook as a file?",
+            "Disconnected", function() {
+              // "Save", save the notebook as a file on the client side
+              bkSessionManager.dumpDisplayStatus();
+              var timeoutPromise = $timeout(function() {
+                bkUtils.saveAsClientFile(
+                    bkSessionManager.getSaveData().notebookModelAsString,
+                "notebook.bkr");
+              }, 1);
+              timeoutPromise.then(function() {
+                $scope.disconnectedDialog = null;
```

Comment:
Can we not use null? JavaScript's null really shouldn't exist and is super weird. For example try doing this in the console:

    typeof null; // it's "object"

The best way to handle this is to set that value to undefined but without using the literal "undefined" since that's just a variable name. The best way to do this is to use void:

    $scope.disconnectedDialog = void 0;

This will give us 100% actual undefined primitive. 

http://stackoverflow.com/questions/1291942/what-does-javascriptvoid0-mean